### PR TITLE
📖 Add script to find files missing in SUMMARY.md

### PR DESCRIPTION
**Pull request by oscr**
**What this PR does / why we need it**:

Adds a script that checks if a filename exists in SUMMARY.md and fixes the findings.

```shell
./util-find-missing-summary.sh 
Didn't find developer/providers/implementers-guide/configure.md in SUMMARY.md
Didn't find SUMMARY.md in SUMMARY.md
Didn't find tasks/external-etcd.md in SUMMARY.md
Didn't find tasks/using-kustomize.md in SUMMARY.md
Didn't find user/guide.md in SUMMARY.md
```

Just a nice to have tool since it's easy to forget adding your file to SUMMARY.md. 

I didn't add SUMMARY.md to SUMMARY.md. Should maybe add a filter for it? :thinking:

### CODE REVIEW

**Review by killianmuldoon — COMMENTED** [review]
This is cool! We should definitely add it to the CI in an action - will be really useful to fix up PRs for docs as they come in.

**Comment by oscr** [comment]
I would love to add it to the workflow! 

Just two questions: 
1) How should SUMMARY.md be handled? Added to itself or just ignored by the script?
1) Should it be a failure or a warning if the script has any findings?

**Comment by killianmuldoon** [comment]
From my POV the simplest thing to do is ignore Summary.md itself and the job should fail if there's any findings.

**Comment by sbueringer** [comment]
Can we maybe just include it like `verify-shellcheck` into our make verify-gen? It's just so much better to use for us then the link checker in that action today :)

**Comment by killianmuldoon** [comment]
Working for me through `make verify` now. Great work @oscr - this is going to be a real headache reduction!

File: hack/verify-capi-book-summary.sh [hunk 1]
```diff
@@ -0,0 +1,34 @@
+#!/bin/bash
+
+# Copyright 2019 The Kubernetes Authors.
```
Comment (ykakarap) [inline hack/verify-capi-book-summary.sh:3]:
```suggestion
# Copyright 2022 The Kubernetes Authors.
```
nit

**Review by ykakarap — COMMENTED** [review]
This is a great addition!!  Thank you :) 

/lgtm

File: hack/verify-capi-book-summary.sh (hunk 1 above)
Comment (oscr) [reply hack/verify-capi-book-summary.sh]: @ykakarap Fixed!

**Comment by sbueringer** [comment]
Thx!
/lgtm
/retest

**Pull request merged**